### PR TITLE
{WIP} Create the review request email template

### PR DESCRIPTION
**Pull request by christo-pr**
[Create Email Template](https://trello.com/c/jO9l30C3/39-create-email-template)

### Context
- Create an email template for the review request action
- Create a Graphql mutation that will trigger this action

### Solution
I created a mutation that will send the email for this action and send all the information to the template

### CODE REVIEW

**Review by bennett-jacob — DISMISSED** [review]

File: src/api/data/scheduledMessage/scheduledMessage.mutations.ts [hunk 1]
```diff
@@ -6,9 +6,14 @@ import { subscriptions } from './scheduledMessage.subscriptions'
...
 import * as moment from 'moment'
 
+// Test purposes
+import sendMail, { IMailerHTML } from '../../../config/mailer'
+import { templates } from '../../../config/mailer/templateParser'
+
 const Mutation = gql`
   extend type Mutation {
     createScheduledMessage(messageData: ScheduledMessageInput!): ScheduledMessage
+    reviewRequest(emailData: ReviewRequestData): Boolean
```
Comment (bennett-jacob) [inline src/api/data/scheduledMessage/scheduledMessage.mutations.ts]: Where is `reviewRequestData` coming from?

Comment (bennett-jacob) [reply src/api/data/scheduledMessage/scheduledMessage.mutations.ts]: jk, found it.

Comment (christo-pr) [reply src/api/data/scheduledMessage/scheduledMessage.mutations.ts]: the type is defined on `src/api/data/scheduledMessage/scheduledMessage.type.ts`

File: src/api/data/scheduledMessage/scheduledMessage.mutations.ts [hunk 2]
```diff
@@ -36,5 +41,23 @@ export const mutationResolvers = {
       pubsub.publish(subscriptions.SCHEDULEDMESSAGE_ADDED, scheduledMessage)
       return scheduledMessage
     },
+    async reviewRequest(obj, { messageData }, context, info) {
+       const data: IMailerHTML<any> = {
+        to: 'tpll5wiiajplpxd6@ethereal.email',
```
Comment (bennett-jacob) [inline src/api/data/scheduledMessage/scheduledMessage.mutations.ts]: This should come from function args. Add a field for the to email in `ReviewRequestData`.

File: src/config/mailer/templates/reviewRequest/reviewRequest.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface IReviewInvite {
```
Comment (bennett-jacob) [inline src/config/mailer/templates/reviewRequest/reviewRequest.ts:1]: add `toEmail: string`

File: src/api/data/scheduledMessage/scheduledMessage.type.ts [hunk 4]
```diff
@@ -15,6 +15,9 @@ const ScheduledMessage = gql`
     recipientEmail: String!
     recipientName: String
   }
+  input ReviewRequestData {
+    link: String!
```
Comment (bennett-jacob) [inline src/api/data/scheduledMessage/scheduledMessage.type.ts:19]: add `toEmail: String!`

File: src/api/data/scheduledMessage/scheduledMessage.mutations.ts (hunk 2 above)
Comment (christo-pr) [reply src/api/data/scheduledMessage/scheduledMessage.mutations.ts]: Ok! I'll include on the params and in the interface

**Review by bennett-jacob — DISMISSED** [review]

**Pull request closed without merging**